### DISCUSSION TITLE
Run a query prior to any query

### DISCUSSION THREAD

**Discussion by francoisneron** [grafana/grafana-infinity-datasource#570] [Q&A]
I have a secure API that has always a two-step process:

1. Get a session token from my personal token (bearer token) - I'm able to get the session token from the JSON response. It's like GET {{host}}/getsessiontoken
2. I use that session token (as a Bearer auth) in all my other calls/endpoints afterwards, and it always expires after 15m or so.
POST {{host}}/serviceX 

How can I have a "required" or automate the first call in the Infinity plugin data source?

**Comment by astro-manny** [comment]
Hello @francoisneron , did you find a solution on sequencing your infinity datasource query calls?

I have a similar challenge. I want to block the execution of two POST queries when the containing panel loads. Then, I want to run each query on-demand and in sequence, without any assumptions about their in between timing. The first datasource query needs to fire manually, and wait until the POST response is evaluated before firing the second datasource query manually. Blocking query execution on panel load is really where my issue lies.

This Question is the closest one to my issue before opening a new one...
@yesoreyeram Is this a grafana question, or does infinity accommodate such use-case? Amazing plugin by the way 🙏
Thank you in advance! Manny

**Comment by Yoganand0708** [comment]
Hi @francoisneron, Greetings!! Did you get any breakthrough on how to implement this using Infinity.

I have a similar situation and want to check with you on this before creating a new case.  

@yesoreyeram any thoughts on handling this in Grafana.. Thanks a lot in advance.

Regards,
Yoganand
